src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -320,6 +326,37 @@ private void Init(string path, FileMode mode, FileAccess access, FileShare share
             }
         }
 
+        private bool CanLockTheFile(Interop.Sys.LockOperations lockOperation, FileAccess access)
+        {
+            Debug.Assert(lockOperation == Interop.Sys.LockOperations.LOCK_EX || lockOperation == Interop.Sys.LockOperations.LOCK_SH);
+
+            if (DisableFileLocking)
+            {
+                return false;
+            }
+            else if (lockOperation == Interop.Sys.LockOperations.LOCK_EX)
+            {
+                return true; // LOCK_EX is always OK
+            }
+            else if ((access & FileAccess.Write) == 0)
+            {
+                return true; // LOCK_SH is always OK when reading
+            }
+
+            switch (Interop.Sys.GetFileSystemType(this))
+            {
+                case 0x6969: // NFS_SUPER_MAGIC
+                case 0xFF534D42: // CIFS_MAGIC_NUMBER
+                case 0x517B: // SMB_SUPER_MAGIC
+                case 0xFE534D42: // SMB2_SUPER_MAGIC (#53182)
```

Comment:
Can you please define the allowed file system types in consts or an enum?

Reply:
It'd be nice to refactor https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.FormatInfo.cs#L33-L146 so we're not duplicating these consts.